src/request/util.ts

```diff
@@ -0,0 +1,24 @@
+import { RequestOptions } from '../request';
+import UrlSearchParams from '../UrlSearchParams';
+
+/**
+ * Returns a URL formatted with optional query string and cache-busting segments.
+ *
+ * @param url The base URL.
+ * @param options The options hash that is used to generate the query string.
+ */
+export function generateRequestUrl(url: string, options: RequestOptions): string {
+	let query = options.query || '';
+
+	if (typeof query === 'object') {
+		query = new UrlSearchParams(query).toString();
+	}
+
+	if (options.cacheBust) {
+		const cacheBust = String(Date.now());
+		query += query ? '&' + cacheBust : cacheBust;
+	}
+
+	const separator = url.indexOf('?') > -1 ? '&' : '?';
```

Comment:
Currently, `URLSearchParams` will take a full URL and return a full URL. This is the wrong behavior according to the spec (which I just checked), so I thought we could use that to our advantage. Leave this line. Nice work on removing the redundant logic :).